src/main/java/com/baidu/hugegraph/threadpool/PausableScheduledThreadPool.java

```diff
@@ -43,20 +43,22 @@ public PausableScheduledThreadPool(int corePoolSize,
         super(corePoolSize, factory);
     }
 
-    public synchronized void pauseSchedule() {
+    public void pauseSchedule() {
         this.paused = true;
         LOG.info("PausableScheduledThreadPool was paused");
     }
 
-    public synchronized void resumeSchedule() {
+    public void resumeSchedule() {
         this.paused = false;
-        this.notifyAll();
+        synchronized (this) {
```

Comment:
move `this.paused = false;` into synchronized block to avoid race between resumeSchedule and resumeSchedule